packages/core/src/components/fields/ObjectField.js

```diff
@@ -214,14 +214,7 @@ class ObjectField extends Component {
     const { SchemaField, TitleField, DescriptionField } = fields;
     const schema = retrieveSchema(this.props.schema, rootSchema, formData);
 
-    // If this schema has a title defined, but the user has set a new key/label, retain their input.
-    let title;
-    if (this.state.wasPropertyKeyModified) {
-      title = name;
-    } else {
-      title = schema.title === undefined ? name : schema.title;
-    }
-
+    const title = schema.title === undefined ? name : schema.title;
```

Comment:
Not specific to these changes... You might want to make this falsey rather than specifically checking for undefined, in case a schema.title is set as false or null. I admit, that would be weird, but I think you'd still want to default to name there.

Reply:
Hmm, not quite sure about that. Sometimes it's useful for me to set schema.title = "" in order to show _no title_ (I know there's also a uiSchema option for hiding titles, but perhaps it would make sense if the schema title is set to a falsey value, just to not show a title in that case)?